src/scheduler/data_types.h

```diff
@@ -673,6 +673,7 @@ struct node_info
 	char *mom;			/* host name on which mom resides */
 
 	char **jobs;			/* the name of the jobs currently on the node */
+	std::unordered_set<std::string> msvr_rmt_jobs;	/* ids of remote jobs running on this node in a msvr setup */
```

Comment:
The main reason why I used the set is that there can be multiple remote jobs running on a vnode, so string matching would be much faster when using unordered sets. But, unfortunately, what you said is also true, since a job can occupy multiple slots, there can be multiple of the same jobid in msvr_rmt_jobs, just like the "jobs" attribute. To optimize this would require creating a separate job info structure just for this, so I didn't think it was worth it since jobs won't span that often. Once we convert server to CPP, we can use sets there as well to create a unique jobid string. Let me know what you think.

Reply:
I don't know if I agree that jobs won't span often.  In HTC, sure, they won't span.  In HPC systems, there will be large multi-node jobs.  We'll try and keep them local, but I don't know if that'll be possible in the average case.  In the average case, a system is mostly full, so we'll have to grab nodes where we can get them.

Reply:
Ok, I agree, but I think that a performance hit because of possibly repetitive jobids inside an attribute wouldn't matter much if a site's system is mostly full .. I guess I'm trying to weasel my way out of solving this, but I really would rather not solve it now if it isn't a big deal because it can be solved much more easily once we move to cpp.

Reply:
I'm not sure you can expect the rest of PBS to be moved to C++ any time soon.  In any case, I haven't reviewed this PR, but why does the scheduler care of the same job shows up multiple times on this line?  I mean the jobs line is converted down to a pointer array of each job being on the node once.  How is using a set a problem?

Reply:
we match jobids that we got from the jobs attribute with the jobids that are present on the msvr_rmt_jobs attribute to rule out any remote jobs as potential ghost jobs. The unordered set is just to speed this matching up, converts an O(N) walk to O(1), that's all.

Reply:
OK, I was just confused.  I didn't see any problems using an unordered_set, and it appears there are none :)